### ISSUE TITLE
Allow enums as types

### ISSUE THREAD

**Issue by cryptoisgood** [demergent-labs/azle#250]
enums in typescripts are strings but it seems azle doesn't like them

``` 
Could not generate Candid record for type alias declaration: undefined
```

Works:
``` 
export type ResponseDto = Variant<{
    Ok?: Opt<TokenMetadata>,
    Err?: Opt<string>
}>;
```

Doesn't work : 
```
export type ResponseDto = Variant<{
    Ok?: Opt<TokenMetadata>,
    Err?: Opt<NftError>
}>;

export enum NftError {
    UnauthorizedOwner = "UnauthorizedOwner",
    UnauthorizedOperator = "UnauthorizedOperator",
    OwnerNotFound = "OwnerNotFound",
    OperatorNotFound = "OperatorNotFound",
    TokenNotFound = "TokenNotFound",
    ExistedNFT = "ExistedNFT",
    SelfApprove = "SelfApprove",
    SelfTransfer = "SelfTransfer",
    TxNotFound = "TxNotFound",
}```

**Comment by lastmjs** [comment]
You need to use Variants, Variant is the equivalent of Enum for Azle applications. I don't think it's desirable to enable Enums as Variants are the equivalent but more powerful type in Rust, Motoko, and Candid.

A number of issues make the TypeScript enum undesirable to use and hard to make work with Candid, including that TypeScript enum variants/branches at runtime are numbers, and TypeScript enums cannot contain values.

Basically Variant should be embraced, it is the new Enum.

**Comment by lastmjs** [comment]
See algebraic data types like in Rust and Motoko for more information: https://en.m.wikipedia.org/wiki/Algebraic_data_type

**Comment by cryptoisgood** [comment]
Indeed, I was actually using enums as constants but I understand your point that it be weird to serialize since it's not actually a string. 

So I'll just regular constants like a normal person.

**Comment by lastmjs** [comment]
BTW I'm getting rid of the need for `?` when defining the properties of `Variant`s, so that's a nice step up.
